charmcraft/application/commands/remote.py

```diff
@@ -0,0 +1,196 @@
+"""Build a charm remotely on Launchpad."""
+import argparse
+import os
+import pathlib
+import time
+from collections.abc import Collection
+from typing import Any, cast
+
+from craft_application.commands import ExtensibleCommand
+from craft_application.launchpad.models import Build, BuildState
+from craft_application.remote.utils import get_build_id
+from craft_cli import emit
+from overrides import override  # pyright: ignore[reportUnknownVariableType]
+
+from charmcraft import models, utils
+
+OVERVIEW = """\
+Command remote-build sends the current project to be built
+remotely.  After the build is complete, packages for each
+architecture are retrieved and will be available in the
+local filesystem.
+
+Interrupted remote builds can be resumed using the --recover
+option, followed by the build number informed when the remote
+build was originally dispatched. The current state of the
+remote build for each architecture can be checked using the
+--status option.
+
+To set a timeout on the remote-build command, use the option
+``--launchpad-timeout=<seconds>``. The timeout is local, so
+the build on launchpad will continue even if the local instance
+is interrupted or times out.
+"""
+
+_CONFIRMATION_PROMPT = (
+    "All data sent to remote builders will be publicly available. "
+    "Are you sure you want to continue?"
+)
+
+
+class RemoteBuild(ExtensibleCommand):
+    """Run analysis on a built charm."""
+
+    name = "remote-build"
+    help_msg = "Build a charm remotely on Launchpad."
+    overview = OVERVIEW
+    always_load_project = True
+
+    @override
+    def _fill_parser(self, parser: argparse.ArgumentParser) -> None:
+        parser.add_argument("--recover", action="store_true", help="recover an interrupted build")
+        parser.add_argument(
+            "--launchpad-accept-public-upload",
+            action="store_true",
+            help="acknowledge that uploaded code will be publicly available.",
+        )
+        parser.add_argument(
+            "--launchpad-timeout",
+            type=int,
+            default=0,
+            metavar="<seconds>",
+            help="Time in seconds to wait for launchpad to build.",
+        )
+        parser.add_argument(
+            "--project-dir",
+            type=pathlib.Path,
+            default=None,
+            help="set the directory for the project",
+        )
+
+    def _run(self, parsed_args: argparse.Namespace, **kwargs: Any) -> int | None:  # noqa: ANN401
+        """Run the remote-build command.
+
+        :param parsed_args: parsed argument namespace from craft_cli.
+
+        :raises AcceptPublicUploadError: If the user does not agree to upload data.
+        """
+        if os.getenv("SUDO_USER") and os.geteuid() == 0:
+            emit.progress(
+                "Running with 'sudo' may cause permission errors and is discouraged.",
+                permanent=True,
+            )
+            time.sleep(1)
```

Comment:
Why sleep here

Reply:
To give a bit of a pause before moving on. I suppose we could do the same "confirm with user" dance with the Launchpad uploads as well and offer an obnoxiously long CLI parameter so people won't just add it every time they run. @mr-cal what do you think?

Reply:
I imagine the UX is a little nicer with this pause and 1 second is small compared to the overall time for a remote build, so I'm fine with this.

A comment explaining the delay is for UX would be nice.

Reply:
Comment added!